ironic/drivers/modules/drac/raid.py

```diff
@@ -996,6 +1064,44 @@ def _create_virtual_disks(task, node):
     return _commit_to_controllers(node, controllers)
 
 
+def controller_supports_ehba_mode(settings, controller_fqdd):
+    # check the raid controller mode
+    raid_controller_mode = settings.get(
+        "{}:RAIDCurrentControllerMode".format(controller_fqdd))
+    raid_controller_mode = raid_controller_mode.current_value
+
+    # if e-HBA found then return True
+    if "Enhanced HBA" in raid_controller_mode:
+        return True
+    else:
+        return False
+
+
+def _config_raid_settings(node):
+    raid_settings = list_raid_settings(node)
+    controllers = list_raid_controllers(node)
+    raid_controllers = [cntrl for cntrl in controllers
+                        if cntrl.model.startswith('PERC H740P')
```

Comment:
If you make the above change to determine if a RAID controller support HBA mode, then you should be able to remove the check on the model number above, I think, though you will need to try the code against a node with BOSS+HBA and a node with a PERC H730P.

Reply:
Sure, i will test this change against node with BOSS+HBA and node with PERC H730P. Will update then after.

Reply:
Done